client/my-sites/domains/domain-management/transfer/transfer-out/select-ips-tag.jsx

```diff
@@ -0,0 +1,225 @@
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import Dialog from 'components/dialog';
+import SearchCard from 'components/search-card';
+import SectionHeader from 'components/section-header';
+import Suggestions from 'components/suggestions';
+import FormButton from 'components/forms/form-button';
+import { saveDomainIpsTag } from 'state/domains/transfer/actions';
+import {
+	getGainingRegistrar,
+	isIpsTagSaving,
+	isTransferDone,
+} from 'state/selectors';
+
+const debug = debugFactory( 'calypso:domains:select-ips-tag' );
+
+class SelectIpsTag extends Component {
+	static ipsTagListUrl = 'https://widgets.wp.com/domains/ips-tag-list.min.json';
+
+	state = {
+		currentQuery: '',
+		ipsTagInput: '',
+		ipsTagList: [],
+		selectedRegistrar: {},
+		showDialog: false,
+	};
+
+	componentDidMount() {
+		request.get( SelectIpsTag.ipsTagListUrl ).then( ( response ) => {
+			this.receiveIpsTagList( response.body );
+		} )
+		.catch( ( error ) => {
+			debug( 'Failed to load IPS tag list! ' + error );
+		} );
+	}
+
+	componentDidUpdate() {
+		if ( this.state.currentQuery && ( this.state.currentQuery !== this.state.ipsTagInput ) ) {
+			this.hideSuggestions();
+		}
+	}
+
+	receiveIpsTagList = ipsTagList => {
+		this.setState( { ipsTagList } );
+	}
+
+	handleKeyDown = event => this.suggestionsRef.handleKeyEvent( event );
+
+	handleSearchInputChange = query => this.setState( { currentQuery: query, ipsTagInput: query } );
+
+	handleSuggestionClick = position => {
+		const parsedLabel = position.label.split( ' ' );
+		this.setState( { ipsTagInput: parsedLabel[ 0 ] } );
+	}
+
+	handleSubmit = () => {
+		this.props.saveDomainIpsTag( this.props.selectedDomainName, this.state.selectedRegistrar );
+	}
+
+	getRegistrarInfo( ipsTag, ipsTagList ) {
+		return find( ipsTagList, [ 'tag', ipsTag ] );
+	}
+
+	getSuggestions() {
+		return this.state.ipsTagList
+			.filter( hint => this.state.currentQuery && startsWith( hint.tag, toUpper( this.state.currentQuery ) ) )
+			.map( hint => ( { label: hint.tag + '  (' + hint.registrarName + ')' } ) );
+	}
+
+	setSuggestionsRef = ref => ( this.suggestionsRef = ref );
+
+	hideSuggestions = () => this.setState( { currentQuery: '' } );
+
+	popOverDialog = () => {
+		const { ipsTagInput, ipsTagList } = this.state;
+		let selectedRegistrar = this.getRegistrarInfo( ipsTagInput, ipsTagList );
+
+		if ( isEmpty( selectedRegistrar ) ) {
+			selectedRegistrar = { tag: toUpper( ipsTagInput ), registrarName: '', registrarUrl: '' };
+		}
+
+		this.setState( {
+			selectedRegistrar,
+			showDialog: true,
+		} );
+	}
+
+	onCloseDialog = ( action ) => {
+		if ( 'submit' === action ) {
+			this.handleSubmit();
+		}
+
+		this.setState( { showDialog: false } );
+	}
+
+	renderGoToGainingRegistrar() {
+		const { translate, gainingRegistrar: { registrarUrl } } = this.props;
+
+		return (
+			<Card
+				className="transfer-out__select-ips-tag"
+				href={ registrarUrl }
+				target="_blank"
```

Comment:
`rel="noopener noreferrer"` - https://dev.to/ben/the-targetblank-vulnerability-by-example